x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/eql/build_alert_group_from_sequence.ts

```diff
@@ -195,14 +201,23 @@ export const objectArrayIntersection = (objects: object[]) => {
   }
 };
 
-export const objectPairIntersection = (a: object | undefined, b: object | undefined) => {
-  if (a === undefined || b === undefined) {
+/**
+ *
+ * @param accumulated
+ * @param child
+ * @returns
```

Comment:
I don't think `accumulated` and `child` are accurate variable names here, it's really just two objects:

```suggestion
 * Finds the intersection of two objects by recursively finding the "intersection" of each of of their common keys' values. If an intersection cannot be found between a key's values, the value will be undefined in the returned object.
 * @param firstObject 
 * @param secondObject
 * @returns an object with all intersecting keys/values, or undefined if there were none.
```

Reply:
> it's really just two objects

True - I changed the names while tracing through the code since this function is called inside of reduce with `acc, obj` as the parameters so I wanted to carry that through. I can change it back to object a and object b though.